pkg/auth/user_auth.go

```diff
@@ -44,6 +47,12 @@ func (u *UserAuthClient) Authenticate(ctx context.Context) (user model.User, aut
 		case ErrMFARequired:
 			u.mfaTypes = nil
 			for _, choiceType := range resp.Data.Choices {
+				if _, ok := unsupportedMfaTypes[choiceType]; ok {
+					logger.Infof("User %s login need MFA, skip %s as it not supported", u.Opts.Username,
+						choiceType)
+					continue
+				}
+
 				u.authOptions[choiceType] = authOptions{
 					MFAType: choiceType,
 					Url:     resp.Data.Url,
```

Comment:
There don't appear to be any syntax errors or missing information in this code snippet. However, there is one difference between the commented lines and uncommented ones:

In the commented line, `unsupportedMfaTypes := map[string]bool{}`, but later it's defined again on line 54 as:

`unsupportedMfaTypes := [...]`

To improve readability and maintainability of the code, I would recommend removing the unneeded declaration and just using default empty values.

The rest looks good; no obvious issues were found, however if you have additional concerns like performance improvements, changes that could simplify complex logic etc., please tell me so!